tentacoli.js

```
@@ -291,6 +294,21 @@ Tentacoli.prototype.request = function (data, callback) {
   return this
 }
 
+Tentacoli.prototype.fire = function (data, callback) {
+  var that = this
+  var req = new Request(null, callback)
+
+  try {
+    wrapStreams(that, data, req, true)
+  } catch (err) {
+    return this
```

this needs a comment.

---

What do you mean?

---

we are swallowing an error. This is a very bad practice.

---

Yeah, I understood that, I didn't catch how you suggest to handle it.

---

There are two options (both valid):

a. remove the `try/catch`, and blow up the exception to the user
b. add a comment here why we are swallowing the exception

The safest option might be a, but b. is also ok.

---

Personally I'll go for `b`, for two reasons:
- The user does not need to write try catch block every time, very annoying.
- The function is a fire and forget, this means that the user do not care about the result, otherwise it should use a standard request.

Thoughts?

---

`wrapStreams` throws if it is a developer error.

---

Otherwise we could accept a callback and use it just for user errors.

```js
sender.fire('message', err => { if (err) console.log(err) })
```
In this way we demand to the user the error control, and if the user does not give us a callback we'll use a `noop`.

---

I'm ok with that. Pass the callback to `writeToStream` as well, in that case it can be used for control flow and backpressure.